ApiController.php

```diff
@@ -16,12 +16,14 @@
     public function doAction(CustomerReposiotry customer)
     {
 	// What kind of problem can we have when using global variables?
+//Concurrency issues//Testing and Confinement 
```

Comment:
Concurrency issues? Can you explain?

Reply:
Yes Concurrency issues .If using  globals variable can be access by multiple threads of execution, synchronization is necessary. When dynamically linking modules with globals, the composed system might not be thread-safe even if the two independent modules tested in dozens of different contexts were safe.